Buildings/Controls/OBC/CDL/Integers/Sources/TimeTable.mo

```diff
@@ -0,0 +1,207 @@
+within Buildings.Controls.OBC.CDL.Integers.Sources;
+block TimeTable "Table look-up with respect to time with constant segments"
+
+  parameter Real table[:,:]
+    "Table matrix with time as a first table column (in seconds, unless timeScale is not 1) and Integers in all other columns";
+  parameter CDL.Types.Extrapolation extrapolation=CDL.Types.Extrapolation.HoldLastPoint
+    "Extrapolation of data outside the definition range";
```

Comment:
Remove extrapolation option, see comment.

Reply:
Which comment are you referring to? If I remove extrapolation, how should the values outside of the time range be handled.

Reply:
I see, you want it to be periodic. That's probably good for now, yet, there is a way to implement it without looking a the derivative, you'd say equal to first point when time less than firs point in time, and equivalent for after the last point.